hibernate-core/src/test/java/org/hibernate/orm/test/mapping/collections/OrderColumnListIndexBaseTest.java

```diff
@@ -100,18 +125,30 @@ else if ( id == 3 ) {
 	}
 
 	@Entity(name = "Person")
+	@Cacheable
+	@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
 	public static class Person {
 
 		@Id
 		private Long id;
 
 		//tag::collections-customizing-ordered-list-ordinal-mapping-example[]
-		@OneToMany(mappedBy = "person", cascade = CascadeType.ALL)
+		@OneToMany(fetch = FetchType.EAGER, mappedBy = "person", cascade = CascadeType.ALL)
 		@OrderColumn(name = "order_id")
 		@ListIndexBase(100)
 		private List<Phone> phones = new ArrayList<>();
 		//end::collections-customizing-ordered-list-ordinal-mapping-example[]
 
+		@ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
+		@JoinTable(name = "parent_child_relationships", joinColumns = @JoinColumn(name = "parent_id"), inverseJoinColumns = @JoinColumn(name = "child_id"))
+		@OrderColumn(name = "pos")
+		@ListIndexBase(200)
+		private List<Person> children = new ArrayList<>();
+
+		@ManyToOne
+		@JoinColumn(name = "mother_id")
+		private Person mother;
```

Comment:
Not sure why you need this to-one in this test, if it's superfluous please remove it.

Reply:
@mbladel I know I uglified the data model of this test case. My aim was to create a reproducer for the HHH-18771 which I found to be kind of complicated - I had to do a number of try-error cycles to reproduce this. I did not find a speaking, canonical straight forward way to get test coverage on `ListInitializer#resolveInstanceSubInitializers`. Additionally I believe that also `ArrayInitializer#resolveInstanceSubInitializers` is affected by the issue but I could not manage to cover this at all. I guess finally you hibernate devs should have deeper knowledge required to create a straight forward test covering both mentioned methods and reproducing the issue. Consider my changes as kind of heuristic. If I remove this Attribute `mother` or the EAGER loading on on the `@OneToMany` collection above it won't reproduce and I don't know why.

Reply:
Since this changes may affect the original test class, producing different results, please create a separate one which only tests the problem described in the issue you're trying to fix and ensures no regressions will be introduced in the future. I suggest trying to only include properties which are needed to reproduce the bug, unless you really need this additional to-one and / or the `EAGER`ness of another association (don't really see why, but I would need to check).

Reply:
Hey @Selaron any news on the separate test? I can try taking a look at your reproducer if you don't have the time to dedicate to it, the fix is easy and looks correct so it would be nice to have.